Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,237 @@
+    {
+        if (collision.gameObject.tag == slowTerrainTag)
+            isInSlowTerrain = !isInSlowTerrain;
+    }
+
+    /// <summary>
+    /// Gets player input
+    /// </summary>
+    private void GetInput()
+    {
+        movementInput = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"));
+        if (movementInput.magnitude > 0)
+            hasReceivedMovementInput = true;
+        else
+            hasReceivedMovementInput = false;
+    }
+
+    /// <summary>
+    /// Adjusts the player to fit exactly into a tile
+    /// </summary>
+    private void SnapToGrid()
+    {
+        int positionX = Mathf.RoundToInt(transform.position.x);
+        int positionY = Mathf.RoundToInt(transform.position.y);
+        transform.position = new Vector3(positionX, positionY);
+    }
+
+    /// <summary>
+    /// Triggers the player's animations based on movement direction
+    /// </summary>
+    /// <param name="movementDirection">What direction is the player moving/facing?</param>
+    /// <param name="slowTerrain">Is the player in slow terrain?</param>
+    /// <param name="cannotMove">Is the player able to move?</param>
+    private void PlayAnim(directions movementDirection, bool slowTerrain, bool cannotMove)
+    {
+        playerAnimator.SetBool("Slow", slowTerrain);
+        playerAnimator.SetBool("Idle", cannotMove);
+        switch (movementDirection)
+        {
+            case directions.Down:
+                playerAnimator.SetTrigger("Move Down");
+                break;
+            case directions.Up:
+                playerAnimator.SetTrigger("Move Up");
+                break;
+            case directions.Left:
+                playerAnimator.SetTrigger("Move Left");
+                break;
+            case directions.Right:
+                playerAnimator.SetTrigger("Move Right");
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Raycasts for collision in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is something in the way, else false.</returns>
+    private bool IsCollisionBlockingMovement(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection,1f, collisionLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Raycasts for slow terrain in the current direction before moving
+    /// </summary>
+    /// <returns>True if there is slow terrain in the way, else false.</returns>
+    private bool IsEnteringSlowTerrain(directions attemptedMoveDirection)
+    {
+        RaycastHit2D hit = Physics2D.Raycast(collisionRaycastOrigin.position, normalizedMovementDirection, 1f, slowTerrainLayerMask);
+        return hit.collider != null;
+    }
+
+    /// <summary>
+    /// Removes the less influential movement axis and normalizes what's left
+    /// </summary>
+    private void NormalizeMovementVector()
```

Comment:
Rename to "Getblahblah" and return a vector 2. Also, try to come up with a name that implies you are "removing the less influential movement vector" so that it's clear this is not just vector2.normalize